docs/src/main/sphinx/functions/json.rst

```diff
@@ -2,6 +2,1152 @@
+
+- input conversion errors (malformed JSON input or JSON parameter)
+- JSON path evaluation errors (e.g. division by zero)
+- returned scalar not convertible to the desired type
+
+Examples
+^^^^^^^^
+
+Let ``customers`` be a table containing two columns: ``id:bigint``,
+``description:varchar``.
+
+========== ==================================================
+id         description
+========== ==================================================
+101        '{"comment" : "nice", "kids" : [10, 13, 16]}'
+102        '{"comment" : "problematic", "kids" : [8, 11]}'
+103        '{"comment" : "knows best", "kids" : [2]}'
+========== ==================================================
+
+The following query gets the ``comment`` for each customer as ``char(12)``:
+
+.. code-block:: text
+
+    SELECT id, json_value(
+                          description,
+                          'lax $.comment'
+                          RETURNING char(12)
+                         ) comment
+    FROM customers
+
+========== ================
+id         comment
+========== ================
+101        'nice        '
+102        'problematic '
+103        'knows best  '
+========== ================
+
+The following query gets the first kid's age for each customer as ``tinyint``:
+
+.. code-block:: text
+
+    SELECT id, json_value(
+                          description,
+                          'lax $.kids[0]'
+                          RETURNING tinyint
+                         ) kid
+    FROM customers
+
+========== ================
+id         kid
+========== ================
+101        10
+102        8
+103        2
+========== ================
+
+The following query gets the third kid's age for each customer. In the strict
+mode, this should cause a structural error for the customers who do not have
+the third kid. This error is handled according to the ``ON ERROR`` clause.
+
+.. code-block:: text
+
+    SELECT id, json_value(
+                          description,
+                          'strict $.kids[2]'
+                          DEFAULT 'err' ON ERROR
+                         ) kid
+    FROM customers
+
+========== ================
+id         kid
+========== ================
+101        '16'
+102        'err'
+103        'err'
+========== ================
+
+After changing the mode to lax, the structural error is suppressed, and the
+customers without the third kid produce empty sequence. This case is handled
+according to the ``ON EMPTY`` clause.
+
+.. code-block:: text
+
+    SELECT id, json_value(
+                          description,
+                          'lax $.kids[2]'
+                          DEFAULT 'missing' ON EMPTY
+                         ) kid
+    FROM customers
+
+========== ================
+id         kid
+========== ================
+101        '16'
+102        'missing'
+103        'missing'
+========== ================
+
+.. warning::
+    The following functions and operators are not compliant with the SQL
```

Comment:
empty line above paragraph

Reply:
ok .. what does the warning actually apply to .. everything below?

Reply:
yes, everything below.